doc/tutorials/acrn-dm_QoS.rst

```diff
@@ -0,0 +1,119 @@
+.. _acrn-dm_qos:
+
+Enable QoS based on runC container
+##################################
+ACRN supports Device-Model QoS based on runC container to control
+the SOS resources (CPU, Storage, MEM, NET) by modifying the runC configuration file.
+
+What is QoS
+***********
+Traditionally, Quality of Service (QoS) is the description or measurement
+of the overall performance of a service, such as a `computer network
+<https://en.wikipedia.org/wiki/Computer_network>`_ or a `cloud computing
+<https://en.wikipedia.org/wiki/Cloud_computing>`_ service,
+particularly the performance is seen by the users on the network.
+
+What is runC container
+**********************
+Containers are an abstraction at the application layer that packages code
+and dependencies together. Multiple containers can run on the same machine
+and share the OS kernel with other containers, each running as
+isolated processes in user space. runC, a lightweight universal container runtime,
+is a command-line tool for spawning and running containers according
+to the Open Container Initiative (OCI) specification.
+
+ACRN-DM QoS architecture
+************************
+In ACRN-DM QoS design, we run the ACRN-DM in a runC container environment.
+Every time we start a UOS, we run a runC container at first;
+then launch the ACRN-DM in the started container. The resources
+the ACRN-DM QoS can manage for Device-Model:
+
+- CPU utilization
+- Memory amount/limitation
+- I/O bandwidth
+- Network throughput
+
+.. figure:: images/acrn-dm_qos_architecture.png
+   :align: center
+
+   ACRN-DM QoS architecture
+
+ACRN-QoS CPU utilization example
+********************************
+In runC ``config.json`` we set the CPU resource as below for VM0 and VM1:
+
+.. code-block:: none
+
+    "cpu": {
+       "shares": 1024,
+       "quota": 1000000,
+       "period": 500000,
+       "realtimeRuntime": 950000,
+       "realtimePeriod": 1000000,
+       "cpus": "2-3",
+       "mems": "0-7"
+    },
+
+In this example the CPU shares value is 1024, so the VM0 or VM1 device model
+CPU utilization is ``1024 / (1024 + 1024 +1024) = 33%``, which means
```

Comment:
Where does that formula come from (esp. the total of 3x 1024 shares)?

Is there a good document that explains how `cpu.shares` work? I have searched a bit for it but I'm not yet 100% that I understand how it works.

Reply:
Hi  Geoffroy,
  The cpu.shares is the linux cgroup feature.  You can get more info about it in 
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-cpu

Reply:
Add this link into ``cpu.shares``